docs/src/image_loading.md

```diff
@@ -0,0 +1,143 @@
+    2. If Image is coming from flash, read Flash Image Metadata Collection section of Manifest
+
+For every image listed in the Metadata collection:
+
+21. Retrieve the SOC Image (could be Firmware or Configuration payload). MCU RT writes directly the image to the target load address as specified in the image metadata. (In the example custom SOC design, this will be the Vendor RAM or Vendor Cfg Storage)
+22. MCU RT sends a Caliptra mailbox command to authorize the image in the SHA Acc identified by the image_id in the image metadata.
+23. Caliptra RT verifies the computed hash in SHA acc versus the one in the SOC manifest corresponding to the image_id given.
+24. Once verified, Caliptra RT returns Success response in Mailbox.
+25. MCU RT receives Success response.
+
+Steps 21-22, are SOC design-specific options One option is to use the Caliptra 'Go' register to set the corresponding 'Go' wire to allow the target component to process the loaded image.
+21. MCU RT sets the corresponding Go bit in Caliptra register corresponding to the image component.
+22. The Go bit sets the corresponding wire that indicates the component can process the loaded image.
+
+## Architecture
+
+The following diagram presents the software stack architecture where the Image Loading module resides.
+
+<p align="left">
+    <img src="images/image_loading_sw_stack.svg" alt="sw_stack" width="80%">
+</p>
+
+At the top of the stack, the user application interacts with the Image Loading module through high-level APIs. The user application is responsible for initiating the image loading and verification.
+
+The Image Loading module provides the interface to retrieve and parse the manifest from the flash storage, and transfer SOC images from the storage to the target destination.
+
+### Application Interfaces
+
+The APIs are presented as methods of the ImageLoader trait.
+
+```rust
+
+
+/// Trait defining the Image Loading module
+pub trait ImageLoader {
+    /// Retrieves the Image Metadata collection from the image source.
+    /// The ImageLoader module automatically selects the appropriate image source based on the parameter passed by MCU ROM in the HandOff FW table.
+    ///
+    /// # Returns
+    /// - `Ok(ImageMetadataCollection)`: The ImageMetadataCollection if successfully retrieved.
+    /// - `Err(DynError)`: An error if retrieval fails.
+    async fn get_imc(&self) -> Result<ImageMetadataCollection, DynError>;
+
+    /// Loads the specified image to a storage mapped to the AXI bus memory map.
+    /// If the image will be loaded directly to the target component, the AXI mapped load address in the image metadata can be used.
+    ///
+    /// # Returns
+    /// - `Ok()`: Image has been loaded and authorized succesfully.
+    /// - `Err(DynError)`: Indication of the failure to load or authorize the image.
+    async fn load_and_authorize(&self, image_id: u32, address: u64) -> Result<(), DynError>;
```

Comment:
Is there a reason that we define `load_and_authorize` as one function instead splitting into two?

Reply:
I think that is how we typically do it. We generally do both steps as once to reduce back and forth, since we aren't really going to want to do one without the other, right?

Reply:
On one hand, splitting them to 2 might offer some flexibility to the user, but I can't think of a use-case where the user just loads it and not authorize it immediately. I think the advantage of 1 function is just it's simpler to use, and as Chris said reduces back and forth with the application. Also i'd like to prevent them from calling authorize() first before calling load(), so making it atomic just ensures these 2 steps are followed in sequence.

Reply:
I was thinking from the perspective of interface definition, usually one function just does one specific job.  
I agree that reducing back-and-forth interaction with app in our scenario is a valuable point.  Let's keep it as it is.